controllers/add_pods.go

```diff
@@ -226,6 +212,55 @@ func (a AddPods) Reconcile(r *FoundationDBClusterReconciler, context ctx.Context
 	return true, nil
 }
 
+// createInstances builds a new instance and potentially other related
+// resources.
+func createInstance(r *FoundationDBClusterReconciler, context ctx.Context, cluster *fdbtypes.FoundationDBCluster, processClass string, idNum int, configMapHash string, existingMeta *metav1.ObjectMeta) (bool, error) {
+	pod, err := GetPod(cluster, processClass, idNum)
+	if err != nil {
+		r.Recorder.Event(cluster, "Error", "GetPod", fmt.Sprintf("failed to get the PodSpec for %s/%d with error: %s", processClass, idNum, err))
+		return false, err
+	}
+
+	instanceID := GetInstanceIDFromMeta(pod.ObjectMeta)
+
+	if existingMeta != nil && instanceID != GetInstanceIDFromMeta(*existingMeta) {
+		return false, fmt.Errorf("Failed to create new pod to match %s", existingMeta.Name)
+	}
+
+	pod.ObjectMeta.Annotations[LastConfigMapKey] = configMapHash
+
+	services := &corev1.ServiceList{}
```

Comment:
I would move this List request into the if statement below otherwise we query the API server and don't use the response.